include/bredis/impl/connection.ipp

```diff
@@ -134,4 +145,30 @@ Connection<NextLayer>::read(DynamicBuffer &rx_buff) {
     return boost::get<positive_parse_result_t>(parse_result);
 }
 
+template <typename NextLayer>
+template <typename DynamicBuffer>
+positive_parse_result_t
+Connection<NextLayer>::read(DynamicBuffer &rx_buff,
+                            boost::system::error_code &ec) {
+    namespace asio = boost::asio;
+    namespace sys = boost::system;
+
+    auto rx_bytes = asio::read_until(stream_, rx_buff, MatchResult(1), ec);
```

Comment:
You need something like this to make ADL work:
```
using boost::asio::read_until;
auto rx_bytes = read_until(stream_, rx_buff, MatchResult(1), ec);
```

You should double check my code snippet. See:
http://stackoverflow.com/questions/8111677/what-is-argument-dependent-lookup-aka-adl-or-koenig-lookup
http://en.cppreference.com/w/cpp/language/adl